nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -200,7 +200,7 @@ fi
 # Ask the user to set a root password, but only if the passwd command
 # exists (i.e. when mutable user accounts are enabled).
 if [[ -z $noRootPasswd ]] && [ -t 0 ]; then
-    if nixos-enter --root "$mountPoint" -c 'test -e /nix/var/nix/profiles/system/sw/bin/passwd'; then
+    if jq -e '.mutableUsers' "$system/users-groups.json" > /dev/null; then
         set +e
         nixos-enter --root "$mountPoint" -c 'echo "setting root password..." && /nix/var/nix/profiles/system/sw/bin/passwd'
```

Comment:
We might as well ask to set passwords for all `isNormalUser` users in `users-groups.json` without `initialHashedPassword`, `hashedPassword`, `initialPassword`, `password` or `passwordFile` set.

Reply:
I don't think that is in scope for this PR. Didn't the installer used to do that, and are we sure it doesn't still?

Reply:
It only asks password for root. You have to log in as root on tty to set user password after installation.

Reply:
That should probably be a feature ticket and second PR, since it is adding new behavior -- and not necessarily trivial behavior, either, since:

* a reasonable implementation might should also work with subsequently added users, after installation
* it should work in a headless environment without a tty

Reply:
I could try to add this feature after this pull-request is merged.